setup.py

```diff
@@ -41,6 +41,7 @@
         "pyrsistent>=0.14.0",
         "six>=1.11.0",
         "functools32;python_version<'3'",
+        "uritools>=2.2.0",
```

Comment:
I would at some point like to be using [hyperlink](https://pypi.org/project/hyperlink) here. Seems likely that we can make due without using `uritools` if we use it.

Reply:
Hmm, I had looked at this problem a while ago and concluded that uritools was the best tool. Seems that I missed hyperlink. I agree may be a better solution